include/libcyphal/presentation/presentation.hpp

```diff
@@ -0,0 +1,214 @@
+#include "subscriber_impl.hpp"
+
+#include "libcyphal/common/cavl/cavl.hpp"
+#include "libcyphal/transport/errors.hpp"
+#include "libcyphal/transport/msg_sessions.hpp"
+#include "libcyphal/transport/transport.hpp"
+#include "libcyphal/transport/types.hpp"
+#include "libcyphal/types.hpp"
+
+#include <cetl/cetl.hpp>
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <utility>
+
+namespace libcyphal
+{
+namespace presentation
+{
+
+// TODO: docs
+class Presentation final : private detail::IPresentationDelegate
+{
+public:
+    Presentation(cetl::pmr::memory_resource& memory, IExecutor& executor, transport::ITransport& transport) noexcept
+        : memory_{memory}
+        , executor_{executor}
+        , transport_{transport}
+    {
+    }
+
+    /// @brief Makes a message publisher.
+    ///
+    /// The publisher must never outlive this presentation object.
+    ///
+    /// @tparam Message DSDL compiled (aka Nunavut generated) type of the message to publish.
+    /// @param subject_id The subject ID to publish the message on.
+    ///
+    template <typename Message>
+    Expected<Publisher<Message>, transport::AnyFailure> makePublisher(const transport::PortId subject_id)
+    {
+        cetl::optional<transport::AnyFailure> out_failure;
+
+        const auto publisher_existing = publisher_impl_nodes_.search(
+            [subject_id](const detail::PublisherImpl& other_pub) {  // predicate
+                //
+                return other_pub.compareBySubjectId(subject_id);
+            },
+            [this, subject_id, &out_failure]() -> detail::PublisherImpl* {  // factory
+                //
+                return makePublisherImpl({subject_id}, out_failure);
+            });
+        if (out_failure)
+        {
+            return std::move(*out_failure);
+        }
+
+        auto* const publisher_impl = std::get<0>(publisher_existing);
+        CETL_DEBUG_ASSERT(publisher_impl != nullptr, "");
+
+        return Publisher<Message>{publisher_impl};
+    }
+
+    /// @brief Makes a message subscriber.
+    ///
+    /// The subscriber must never outlive this presentation object.
+    ///
+    /// @tparam Message DSDL compiled (aka Nunavut generated) type of the message to subscribe.
+    /// @param subject_id The subject ID to subscribe the message on.
+    ///
+    template <typename Message>
+    Expected<Subscriber<Message>, transport::AnyFailure> makeSubscriber(const transport::PortId subject_id)
+    {
+        cetl::optional<transport::AnyFailure> out_failure;
+
+        const auto subscriber_existing = subscriber_impl_nodes_.search(
+            [subject_id](const detail::SubscriberImpl& other_sub) {  // predicate
+                //
+                return other_sub.compareBySubjectId(subject_id);
+            },
+            [this, subject_id, &out_failure]() -> detail::SubscriberImpl* {  // factory
+                //
+                return makeSubscriberImpl({Message::_traits_::ExtentBytes, subject_id}, out_failure);
```

Comment:
Yes, it would not, but won't end up in such case in this 
```
    template <typename Message>
    Expected<Subscriber<Message>, transport::AnyFailure> makeSubscriber(const transport::PortId subject_id)
    {
```
function in the first place - different "dummy" deserializer will be in use for such specialization. I call it dummy b/c it will do no attempt to deserialize but just pass scattered buffer as is to its callback.

Reply:
But I believe you are right that such `Presentation::makeSubscriber` overload should have two paramets, like this:
```
Expected<Subscriber<void>, transport::AnyFailure> makeSubscriber(subject_id, extent_bytes)
```